### PR TITLE
[XVideos] Correctly set thumbnail image (fixes issue #15978)

### PR DESCRIPTION
**Pull request by parmjitv**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/rg3/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/rg3/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/rg3/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

Updated the extractor for the XVideos.com website to set the thumbnail image using the embedded html5player data - issue #15978.

Cheers,

Parmjit V.

### CODE REVIEW

File: youtube_dl/extractor/xvideos.py [hunk 1]
```diff
@@ -39,6 +39,20 @@ class XVideosIE(InfoExtractor):
     }, {
         'url': 'http://static-hw.xvideos.com/swf/xv-player.swf?id_video=4588838',
         'only_matching': True,
+    }, {
+        'url': 'https://www.xvideos.com/video299069/diesel_sfw_xxx_video',
```
Comment (dstftw) [inline youtube_dl/extractor/xvideos.py]: Remove.

File: youtube_dl/extractor/xvideos.py [hunk 2]
```diff
@@ -58,7 +72,8 @@ def _real_extract(self, url):
             group='title') or self._og_search_title(webpage)
 
         thumbnail = self._search_regex(
-            r'url_bigthumb=(.+?)&amp', webpage, 'thumbnail', fatal=False)
+            r'url_bigthumb=(.+?)&amp', webpage, 'thumbnail', default=None) or self._search_regex(
+            r'html5player\.setThumbUrl\(\'(.+?)\'\);', webpage, 'thumbnail', fatal=False)
```
Comment (dstftw) [inline youtube_dl/extractor/xvideos.py]: Refactor into a single `_search_regex` call.

**Comment by parmjitv** [comment]
@dstftw - Thanks for the review. I have made the requested corrections, please have a look at your convenience. Thanks!

File: youtube_dl/extractor/xvideos.py [hunk 3]
```diff
@@ -58,7 +58,7 @@ def _real_extract(self, url):
             group='title') or self._og_search_title(webpage)
 
         thumbnail = self._search_regex(
-            r'url_bigthumb=(.+?)&amp', webpage, 'thumbnail', fatal=False)
+            r'url_bigthumb=(.+?)&amp|html5player\.setThumbUrl\(\'(.+?)\'\);', webpage, 'thumbnail', fatal=False)
```
Comment (dstftw) [inline youtube_dl/extractor/xvideos.py]: Pass list of regexes.

**Comment by parmjitv** [comment]
@dstftw - Done.  Please let me know if there is anything else left outstanding. Thank you!

**Pull request merged**